src/kms_sink.cpp

```diff
@@ -157,22 +157,19 @@ int KMSSink::configurePipeline(const libcamera::PixelFormat& format) {
           crtc_ = crtc;
           plane_ = plane;
           format_ = format;
-          goto break_all;
+          return 0;
         }
 
         if (plane->supportsFormat(xFormat)) {
           crtc_ = crtc;
           plane_ = plane;
           format_ = xFormat;
-          goto break_all;
+          return 0;
         }
       }
     }
   }
-
-// hack to fix
-break_all:
-
+  
```

Comment:
This needs to change:

https://github.com/kbingham/libcamera/commit/ca20503f6295a6cc25261ad5b975a73c961ae73b

If we look at the libcamera version of the patch more closely... If we hit either of those lines 160 or 167, we want to break out of all loops and continue at line 173... That's why you have to create the separate selectPipeline function... It's the exact same as the goto behaviour, only split into separate functions...

You want to get to the "Using KMS plane" line in the successful case...